tests/unit/workflows/dotnet_clipackage/test_actions.py

```diff
@@ -42,6 +45,24 @@ def test_global_tool_update_failed(self):
         action = GlobalToolInstallAction(self.subprocess_dotnet)
         self.assertRaises(ActionFailedError, action.execute)
 
+    def test_global_tool_parallel(self):
+        executor = ThreadPoolExecutor()
+
+        actions = [
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+            GlobalToolInstallAction(self.subprocess_dotnet),
+        ]
+
+        for action in actions:
+            executor.submit(action.execute)
+
+        executor.shutdown()
```

Comment:
Will this wait till the spawned threads finishes execution? In java, the main thread need to "join" those threads first before releasing the pool.

Reply:
https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown  wait is True by default, so yes

Reply:
You can avoid calling it though if you use `with ThreadPoolExecutor() as executor:`

Reply:
Good one, will update to use context here